Add incompatible status.

cmeiklejohn opened the pull request:
If an RPC call fails to a particular node for 
a reason other than nodedown, mark the node as 
Incompatible with Control, rather than reporting
the node as being Unreachable.

This prevents false alarms during upgrades where
a portion of the cluster may be running a 
pre-1.1 release.

Screenshots (tried embedding, didn't work): 

Overview Page: http://cl.ly/1C0R2F062h0e2I3g2B3I
Ring Page: http://cl.ly/1J2U0C1K1m420v2l2I42
Cluster Page: http://cl.ly/2U3y3s43281i2j3t1A2x

@beerriot 

See #20 for original bug report.

cmeiklejohn committed:
Add incompatible status.

If an RPC call fails to a particular node for 
a reason other than nodedown, mark the node as 
Incompatible with Control, rather than reporting
the node as being Unreachable.

This prevents false alarms during upgrades where
a portion of the cluster may be running a 
pre-1.1 release.

bookshelfdave commented:
Code looks good, I was able to reproduce the given scenario.

Note: If Riak search isn't enabled, the Disabled indicator on the Ring page just shows up grey instead of red.

cmeiklejohn commented:
@metadave I believe that having the disabled/grey indicator is consistent with how Control works with the Unreachable state so I'm inclined to leave it.  See: http://cl.ly/2C0W3B2O3m3c3c0G2C08

bookshelfdave commented:
+1

cmeiklejohn committed:
Ensure record upgrade.

Nodes running an earlier version of Riak will return an incompatible
record, which will need to be upgrade for the application to function.

cmeiklejohn commented:
@metadave @Vagabond Added b5c9881 which ensures that records returned from nodes running an earlier version of Riak are handled correctly and have the additional information added in.

cmeiklejohn committed:
Rework incompatible into the status field.

This prevents us from having to update the record.

cmeiklejohn commented:
Pushed an update to consolidate the incompatible state into the Status field which doesn't require updating the record.

cmeiklejohn commented:
@metadave mind taking another pass?

bookshelfdave commented:
Everything looks good and ran as expected.

cmeiklejohn commented:
thanks!
